src/sql/src/func.rs

```diff
@@ -3161,13 +3161,7 @@ pub static PG_CATALOG_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(|
             params!(Float32) => AggregateFunc::SumFloat32 => Float32, 2110;
             params!(Float64) => AggregateFunc::SumFloat64 => Float64, 2111;
             params!(Numeric) => AggregateFunc::SumNumeric => Numeric, 2114;
-            params!(Interval) => Operation::unary(|_ecx, _e| {
-                // Explicitly providing this unsupported overload
-                // prevents `sum(NULL)` from choosing the `Float64`
-                // implementation, so that we match PostgreSQL's behavior.
-                // Plus we will one day want to support this overload.
-                bail_unsupported!("sum(interval)");
-            }) => Interval, 2113;
+            params!(Interval) => AggregateFunc::SumInterval => Interval, 2113;
```

Comment:
TODO: Do we need to handle this in some way?

Reply:
I don't entirely understand the question. This is the function `oid` and should match what PostgreSQL uses. It looks like you have the correct one here: https://github.com/postgres/postgres/blob/47c98035c6fbff01724b9a646196c92b99faa6bf/src/include/catalog/pg_proc.dat#L6722-L6725

Reply:
Ah, I did not notice the comment. Hmm, it looks like a hacky workaround for this issue: https://github.com/MaterializeInc/materialize/pull/18243. More specifically, we tend to select the incorrect function when types aren't specified when compared to PostgreSQL. If a function isn't overloaded, then we can't pick the wrong one. So by adding a new overload, you're causing our planner to pick the wrong function when types aren't specified. The way to handle this would be to merge the linked PR, which we've been unable to do for over a year.

I think unfortunately, this should be blocked until #18243 is merged.